### PR TITLE
Add support for RGB and add test cases for it

### PR DESCRIPTION
**Pull request by ClaudeZou**
note: I closed pull request #2 since it didn't have parser.js updated.

### CODE REVIEW

File: lib/parser.pegjs [hunk 1]
```diff
@@ -424,13 +427,29 @@ COLOR_SPEC_NAMED
...
  = "{" _ r:CNUM "," _ g:CNUM "," _ b:CNUM "}" _
    { return "{"+r+","+g+","+b+"}"; }
+COLOR_SPEC_RGB2
+ = "{" _ r:CNUM2 "," _ g:CNUM2 "," _ b:CNUM2 "}" _
+   { return "{"+r+","+g+","+b+"}"; }
 COLOR_SPEC_CMYK
  = "{" _ c:CNUM "," _ m:CNUM "," _ y:CNUM "," _ k:CNUM "}" _
    { return "{"+c+","+m+","+y+","+k+"}"; }
 
+CNUM2
```
Comment (physikerwelt) [inline lib/parser.pegjs]: Is that the simples way to check if the input is integer between 0 and 255?

Comment (physikerwelt) [inline lib/parser.pegjs]: probably one could rename that to cnum2 to byte or numbyte?

**Comment by physikerwelt** [comment]
all in all it looks good to me. @cscott @d00rman @gwicke What do you think?

**Comment by physikerwelt** [comment]
This is related to https://phabricator.wikimedia.org/T107766

File: lib/parser.pegjs (hunk 1 above)
Comment (ClaudeZou) [inline lib/parser.pegjs]: As far as I'm aware, there is no built in way to only parse numbers 0 to 255.
As for renaming it, it doesn't really matter.

**Comment by cscott** [comment]
Don't I already have support for RGB colors in my fork of texvcjs?
 --scott
​

**Comment by physikerwelt** [comment]
@cscott : It seems that you ignore uppercase RGB https://github.com/cscott/texvcjs/blob/master/test/en-wiki-formulae.js#L153 and that you only support lowercase rgb. See https://en.wikibooks.org/wiki/LaTeX/Colors#Color_Models

**Comment by cscott** [comment]
https://github.com/cscott/texvcjs/blob/master/lib/parser.pegjs#L406 has the case insensitive flag.

File: lib/parser.pegjs [hunk 2]
```diff
@@ -403,7 +403,8 @@ DEFINECOLOR
    "{" _ name:alpha+ _ "}" _ "{" _
      a:( "named"i _ "}" _ cs:COLOR_SPEC_NAMED { return "{named}" + cs; }
        / "gray"i  _ "}" _ cs:COLOR_SPEC_GRAY  { return "{gray}" + cs; }
-       / "rgb"i   _ "}" _ cs:COLOR_SPEC_RGB   { return "{rgb}" + cs; }
+       / "rgb"    _ "}" _ cs:COLOR_SPEC_RGB1  { return "{rgb}" + cs; }
+       / "RGB"    _ "}" _ cs:COLOR_SPEC_RGB2  { return "{RGB}" + cs; }
```
Comment (physikerwelt) [inline lib/parser.pegjs]: @cscott see the change. In LaTeX the semantics might be case dependent.

**Comment by cscott** [comment]
Ah.  You are confused: we don't use the `xcolor` package.  We use the `color` package.  See http://tex.stackexchange.com/questions/91271/whats-the-difference-between-color-and-xcolor-packages

**Comment by physikerwelt** [comment]
@cscott https://en.wikipedia.org/wiki/Help:Displaying_a_formula#Color gives an example using the uppercase RGB syntax. I could live with dropping support for uppercase RGB, but I think it would be confusing to change the meaning of RGB to actual meaning of rgb.
@d00rman wanted to have full support for the definecolor command. So what should I do?

**Comment by cscott** [comment]
Fixed wiki, pushed/merged https://github.com/cscott/texvcjs/commit/70ef64cce626dec2e9d023c46dc62f1052f26924

**Comment by physikerwelt** [comment]
Already merged in https://github.com/wikimedia/texvcjs/commit/ef77bd4a2b5c5a62a3501a22718325f47cf0d06e

**Pull request closed without merging**